### PR TITLE
#265 enable editing of tags

### PR DESCRIPTION
**Pull request by arizonagroovejet**
Allows editing of tags on the bench edit page.

Modifies `get_tags_from_bench()` to include tag ids. This allows what it returns to be used with select2. 

Modifies the only pre-existing use of `get_tags_from_bench()` output in `benches.php` to account for this. 

Modifies `save_benches()` to first delete all tags for the bench then add whatever tags are passed. I believe I've done this such that the deletion and adding cannot succeed or fail independently of each other, which avoids scenarios such as all the tags are deleted but then the new set of tags aren't added, or the deletion fails but the new set of tags is added and now the bench has multiple instances of one or more tags.

Adds information about old and new tags to the email sent to administrator.

Tested for:
 - Adding N tags,
 - Removing N tags
 - Removing all tags.

This PR assumes that the disabling of the call to `save_tags` in `upload.php` mentioned in #290 was a mistake and makes https://github.com/openbenches/openbenches.org/pull/292 redundant. 

I always use two spaces for indentation and found myself making concious effort to try to adhere to the existing use of tabs for indentation in the code, but maybe I've failed somewhere.

### CODE REVIEW

File: www/mysql.php [hunk 1]
```diff
@@ -1630,16 +1630,30 @@ function get_bench_tag_count($tagText) {
...
-	foreach ($tags as $tagID) {
-		$save_tag = $mysqli->prepare(
-			"INSERT INTO `tag_map` (`mapID`, `benchID`, `tagID`)
-			VALUES                 (NULL,     ?,         ?)");
-		$save_tag->bind_param('ii', $benchID, $tagID);
-		$save_tag->execute();
-		$save_tag->close();
+	$mysqli->begin_transaction();
+	try {
+		$remove_tags = $mysqli->prepare("DELETE FROM `tag_map` WHERE `benchID`=$benchID");
```
Comment (edent) [inline www/mysql.php:1643]: `benchID` should be prepared. So
```
$remove_tags = $mysqli->prepare("DELETE FROM `tag_map` WHERE `benchID`=?);
$remove_tags->bind_param('i', $benchID);
```

**Review by edent — CHANGES_REQUESTED** [review]
Looks good. Will test it this weekend. One prepared statement needs to be addressed.

**Comment by arizonagroovejet** [comment]
Well that's embarrassing. (I have explained to other people that their code is injectable and to use prepared statements, don't trust user input etc.) Sorry about that. In my defence, see comments I've made elsewhere about not being a developer. ;) I'll try and sort this before the weekend. The workflow aspect of what to do next is all new territory for me, but looks like I can just push to my branch I create the PR from.

**Comment by arizonagroovejet** [comment]
I've made another commit to fix the issue you pointed out.

**Comment by edent** [comment]
Tested and working! Many thanks.

**Pull request merged**